chcemvediet/apps/anonymization/cron.py

```diff
@@ -215,6 +218,48 @@ def recognize_attachment():
     else:
         recognize_using_ocr(attachment_normalization)
 
+def anonymize_attachment():
+    attachment_recognition = (AttachmentRecognition.objects
+            .filter(successful=True,
+                    content_type=content_types.ODT_CONTENT_TYPE,
+                    attachment__attachmentanonymization__isnull=True)
+            .first())
+    try:
+        zipfile_attachment = zipfile.ZipFile(StringIO.StringIO(attachment_recognition.content))
+        inforequest = attachment_recognition.attachment.generic_object.branch.inforequest
+        namespace = {u'text': u'urn:oasis:names:tc:opendocument:xmlns:text:1.0'}
+        filename_relative = os.path.join(u'attachment_anonymizations', random_string(10))
+        filename = os.path.join(settings.PROJECT_PATH, u'media/' + filename_relative)
+        with zipfile.ZipFile(filename, u'w') as zf:
+            for f in zipfile_attachment.filelist:
+                content = zipfile_attachment.read(f)
+                if magic.from_buffer(content, mime=True) == content_types.XML_CONTENT_TYPE:
+                    zf.writestr(f, anonymize_xml(inforequest, content, u'.//text:span', namespace))
+                else:
+                    zf.writestr(f, content)
+
+        AttachmentAnonymization.objects.create(
+            attachment=attachment_recognition.attachment,
+            successful=True,
+            file=filename_relative,
```

Comment:
@martinmacko47 Kedze uz mame existujuci subor ulozeny v `attachment_anonymizations` priecinku. Treba ho uz len prepojit. Je takyto postup spravny? 
Ked som do ContentFile posielal anonymizovany obsah, vyhodnotilo to ako obycajny textovy subor, tiez mi to nevedelo vytvorit strukturu `zip` suboru s viacerymi subormi.

Reply:
@martinmacko47 Je potrebne pri uspesnej anonymizacii pridavat nejaku spravu do `debug`?

Reply:
:ok: